gstreamer-sdp/src/auto/m_i_k_e_y_message.rs

```diff
@@ -0,0 +1,166 @@
+
+    match fn {
+        copy => |ptr| gobject_ffi::g_boxed_copy(ffi::gst_mikey_message_get_type(), ptr as *mut _) as *mut ffi::GstMIKEYMessage,
+        free => |ptr| gobject_ffi::g_boxed_free(ffi::gst_mikey_message_get_type(), ptr as *mut _),
+        get_type => || ffi::gst_mikey_message_get_type(),
+    }
+}
+
+impl MIKEYMessage {
+    pub fn new() -> MIKEYMessage {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_mikey_message_new())
+        }
+    }
+
+    //pub fn new_from_bytes(bytes: &glib::Bytes, info: /*Ignored*/&mut MIKEYDecryptInfo, error: /*Ignored*/Option<Error>) -> MIKEYMessage {
+    //    unsafe { TODO: call ffi::gst_mikey_message_new_from_bytes() }
+    //}
+
+    pub fn new_from_caps(caps: &gst::Caps) -> MIKEYMessage {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_mikey_message_new_from_caps(caps.to_glib_none().0))
+        }
+    }
+
+    //pub fn new_from_data(data: &[u8], info: /*Ignored*/&mut MIKEYDecryptInfo, error: /*Ignored*/Option<Error>) -> MIKEYMessage {
+    //    unsafe { TODO: call ffi::gst_mikey_message_new_from_data() }
+    //}
+
+    pub fn add_cs_srtp(&mut self, policy: u8, ssrc: u32, roc: u32) -> bool {
+        unsafe {
+            from_glib(ffi::gst_mikey_message_add_cs_srtp(self.to_glib_none_mut().0, policy, ssrc, roc))
+        }
+    }
+
+    pub fn add_pke(&mut self, C: MIKEYCacheType, data: &[u8]) -> bool {
+        let data_len = data.len() as u16;
+        unsafe {
+            from_glib(ffi::gst_mikey_message_add_pke(self.to_glib_none_mut().0, C.to_glib(), data_len, data.to_glib_none().0))
+        }
+    }
+
+    pub fn add_rand(&mut self, rand: &[u8]) -> bool {
+        let len = rand.len() as u8;
+        unsafe {
+            from_glib(ffi::gst_mikey_message_add_rand(self.to_glib_none_mut().0, len, rand.to_glib_none().0))
+        }
+    }
+
+    pub fn add_rand_len(&mut self, len: u8) -> bool {
+        unsafe {
+            from_glib(ffi::gst_mikey_message_add_rand_len(self.to_glib_none_mut().0, len))
+        }
+    }
+
+    //pub fn add_t(&mut self, type_: MIKEYTSType, ts_value: &[u8]) -> bool {
+    //    unsafe { TODO: call ffi::gst_mikey_message_add_t() }
+    //}
+
+    pub fn add_t_now_ntp_utc(&mut self) -> bool {
+        unsafe {
+            from_glib(ffi::gst_mikey_message_add_t_now_ntp_utc(self.to_glib_none_mut().0))
+        }
+    }
+
+    pub fn base64_encode(&mut self) -> Option<String> {
+        unsafe {
+            from_glib_full(ffi::gst_mikey_message_base64_encode(self.to_glib_none_mut().0))
+        }
+    }
+
+    pub fn find_payload(&self, type_: MIKEYPayloadType, nth: u32) -> Option<MIKEYPayload> {
+        unsafe {
+            from_glib_none(ffi::gst_mikey_message_find_payload(self.to_glib_none().0, type_.to_glib(), nth))
+        }
+    }
+
+    //pub fn get_cs_srtp(&self, idx: u32) -> /*Ignored*/Option<MIKEYMapSRTP> {
```

Comment:
There are still some ignored functions but let's worry about those once the basics are all merged